src/Core/Tools/SendFeatures/Commands/AnonymousSendCommand.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Bit.Core.Exceptions;
+using Bit.Core.Platform.Push;
+using Bit.Core.Tools.Entities;
+using Bit.Core.Tools.Enums;
+using Bit.Core.Tools.Repositories;
+using Bit.Core.Tools.SendFeatures.Commands.Interfaces;
+using Bit.Core.Tools.Services;
+
+namespace Bit.Core.Tools.SendFeatures.Commands;
+
+public class AnonymousSendCommand : IAnonymousSendCommand
+{
+    private readonly ISendRepository _sendRepository;
+    private readonly ISendFileStorageService _sendFileStorageService;
+    private readonly IPushNotificationService _pushNotificationService;
+    private readonly ISendAuthorizationService _sendAuthorizationService;
+
+    public AnonymousSendCommand(
+        ISendRepository sendRepository,
+        ISendFileStorageService sendFileStorageService,
+        IPushNotificationService pushNotificationService,
+        ISendAuthorizationService sendAuthorizationService
+        )
+    {
+        _sendRepository = sendRepository;
+        _sendFileStorageService = sendFileStorageService;
+        _pushNotificationService = pushNotificationService;
+        _sendAuthorizationService = sendAuthorizationService;
+    }
+
+    // Response: Send, password required, password invalid
+    public async Task<(string, bool, bool)> GetSendFileDownloadUrlAsync(Send send, string fileId, string password)
+    {
+        if (send.Type != SendType.File)
+        {
+            throw new BadRequestException("Can only get a download URL for a file type of Send");
+        }
+
+        var (grantAccess, passwordRequired, passwordInvalid) = _sendAuthorizationService.SendCanBeAccessed(send, password);
```

Comment:
🌱 This pattern might need to be adjusted since we're moving to an authorization token model. How might this change in a situation where `send` can't load without a proper claim? (i.e. from a JWT bearer token received in an auth header.)

Reply:
We could use IHttpContextAccessor within this service to be able to check the roles of the user. I think this method could have these pieces refactored:

-  Move the 'validation' pieces to the ISendValidationService to check user input (i.e lines 38-40 and IsNullOrWhiteSpace checks)
- Verification of the password would be moved the `GetSendFileDownloadUrlAsync`
- SendCanBeAccessed would just be concerned with the claims.

Reply:
> We could use IHttpContextAccessor within this service to be able to check the roles of the user.

I'd prefer to use something independent of the execution context (ie. an HTTP request). The controller has access to the `HttpContext`, for example, and could extract the claim.

> Verification of the password would be moved the GetSendFileDownloadUrlAsync

This won't work. The password won't be given to those URLs. Instead, they'll have a bearer token containing a claim authorizing access to the download.

Reply:
I'm going to resolve this and figure out an authentication-compatible API when we move to the auth token flow.